runtime/v2/shim/shim.go

```diff
@@ -148,7 +148,9 @@ func parseFlags() {
 	flag.StringVar(&bundlePath, "bundle", "", "path to the bundle if not workdir")
 
 	flag.StringVar(&addressFlag, "address", "", "grpc address back to main containerd")
-	flag.StringVar(&containerdBinaryFlag, "publish-binary", "containerd", "path to publish binary (used for publishing events)")
+	flag.StringVar(&containerdBinaryFlag, "publish-binary", "",
+		fmt.Sprintf("path to publish binary (used for publishing events), but %s will ignore this flag, please use the %s env", os.Args[0], ttrpcAddressEnv),
```

Comment:
`ttrpcAddressEnv` will be populated when running via containerd (but this message will not be visible to user). If ran manually by user, this variable will be empty.

Possibly printing a deprecation warning would be better here?

Reply:
How about this?
```go
fmt.Sprintf("Deprecated: path to publish binary (used for publishing events), Please use the %s env", os.Args[0], ttrpcAddressEnv)
```